sdk/dotnet/Pulumi.Automation/Commands/LocalPulumiCmd.cs

```diff
@@ -34,103 +99,86 @@ internal class LocalPulumiCmd : IPulumiCmd
-                    CreateNoWindow = true,
-                    UseShellExecute = false,
-                    RedirectStandardError = true,
-                    RedirectStandardOutput = true,
-                },
-            };
-
-            foreach (var arg in completeArgs)
-                proc.StartInfo.ArgumentList.Add(arg);
-
-            foreach (var pair in env)
-                proc.StartInfo.Environment[pair.Key] = pair.Value;
-
-            var standardOutputBuilder = new StringBuilder();
-            proc.OutputDataReceived += (_, @event) =>
+            if (debugCommands)
             {
-                if (@event.Data != null)
-                {
-                    standardOutputBuilder.AppendLine(@event.Data);
-                    onStandardOutput?.Invoke(@event.Data);
-                }
-            };
+                // Required for event log
+                // We add it after the provided env vars to ensure it is set to true
+                env["PULUMI_DEBUG_COMMANDS"] = "true";
+            }
 
-            var standardErrorBuilder = new StringBuilder();
-            proc.ErrorDataReceived += (_, @event) =>
+            return env;
+        }
+
+        private static IEnumerable<string> PulumiArgs(IEnumerable<string> args, EventLogFile? eventLogFile)
+        {
+            // all commands should be run in non-interactive mode.
+            // this causes commands to fail rather than prompting for input (and thus hanging indefinitely)
+            if (!args.Contains("--non-interactive"))
             {
-                if (@event.Data != null)
-                {
-                    standardErrorBuilder.AppendLine(@event.Data);
-                    onStandardError?.Invoke(@event.Data);
-                }
-            };
+                args = args.Concat(new[] { "--non-interactive" });
+            }
 
-            var tcs = new TaskCompletionSource<CommandResult>();
-            using var cancelRegistration = cancellationToken.Register(() =>
+            if (eventLogFile != null)
             {
-                // if the process has already exited than let's
-                // just let it set the result on the task
-                if (proc.HasExited || tcs.Task.IsCompleted)
-                    return;
+                args = args.Concat(new[] { "--event-log", eventLogFile.FilePath });
+            }
 
-                // setting it cancelled before killing so there
-                // isn't a race condition to the proc.Exited event
-                tcs.TrySetCanceled(cancellationToken);
+            return args;
+        }
 
-                try
-                {
-                    proc.Kill();
-                }
-                catch
-                {
-                    // in case the process hasn't started yet
-                    // or has already terminated
-                }
-            });
+        private class EventLogFile : IDisposable
+        {
+            private string _eventLogFile;
```

Comment:
Since we don't modify the file path, we can use get-only property e.g

```csharp
public string FilePath { get; }
```

Reply:
Most .nET i've written was F# so please keep the style comments coming, very helpful.